tardis/plasma/assembly/legacy_assembly.py

```diff
@@ -0,0 +1,39 @@
+from tardis.plasma.assembly.base import PlasmaSolverFactory
+from tardis.plasma.radiation_field import DilutePlanckianRadiationField
+
+
+def assemble_plasma(config, simulation_state, atom_data=None):
+    """
+    Create a BasePlasma instance from a Configuration object
+    and a SimulationState.
+
+    Parameters
+    ----------
+    config : io.config_reader.Configuration
+    simulation_state : model.SimulationState
+    atom_data : atomic.AtomData
+        If None, an attempt will be made to read the atomic data
+        from config.
+
+    Returns
+    -------
+    : plasma.BasePlasma
+
+    """
+    atomic_numbers = simulation_state.abundance.index
+    plasma_solver_factory = PlasmaSolverFactory(
```

Comment:
Should the config really need to be passed twice?

Reply:
this will change in the future - likely. I'm just rewriting all of it.